src/discordcr/rest.cr

```diff
@@ -13,6 +13,14 @@ module Discord
 
     alias RateLimitKey = {route_key: Symbol, major_parameter: UInt64?}
 
+    # Makes a REST request to Discord, with the given *method* to the given
+    # *path*, with the given *headers* set and with the given *body* being sent.
+    # The *route_key* should uniquely identify the route used, for rate limiting
+    # purposes. The *major_parameter* should be set to the guild or channel ID,
+    # if either of those appears as the first parameter in the route.
+    #
+    # This method also does rate limit handling, so if a rate limit is
+    # encountered, it may take longer than usual.
```

Comment:
People who don't know crystal fibers well may think that this method blocking will stop events from being handled. It's only a very vague suggestion.